src/replace_variables.cpp

```diff
@@ -0,0 +1,94 @@
+#include "vector"
+
+#include "ir/visitor.h"
+#include "replace_variables.h"
+
+#include "frontends/common/constantFolding.h"
+#include "frontends/common/resolveReferences/resolveReferences.h"
+#include "frontends/p4/createBuiltins.h"
+#include "frontends/p4/directCalls.h"
+#include "frontends/p4/simplify.h"
+#include "frontends/p4/simplifyDefUse.h"
+#include "frontends/p4/unusedDeclarations.h"
+
+namespace P4PRUNER {
+
+const IR::Node *ReplaceVariables::postorder(IR::MethodCallExpression *s) {
+    return s;
+}
+const IR::Node *ReplaceVariables::postorder(IR::Expression *s) {
+    auto expr = getOriginal<IR::Expression>();
+    auto type = typeMap->getType(expr, true);
+
+    if (typeMap->isLeftValue(expr)) {
+        return s;
+    }
+    int bits = type->width_bits();
+
+    auto decision = get_rnd_pct();
+    if (decision < 0.5 && type->is<IR::Type_Bits>()) {
+        auto new_elt = new IR::Constant(new IR::Type_Bits(bits, false), 10);
+        return new_elt;
+    }
+    return s;
+}
+
+const IR::P4Program *apply_replace(const IR::P4Program *program,
+                                   P4PRUNER::PrunerConfig pruner_conf) {
+    P4::ReferenceMap refMap;
+    P4::TypeMap typeMap;
+    const IR::P4Program *temp;
+
+    std::initializer_list<Visitor *> passes;
+
+    PassManager pass_manager(passes);
+
+    pass_manager.addPasses({new P4::ResolveReferences(&refMap, true),
+                            new P4::TypeInference(&refMap, &typeMap, false)});
+
+    // if (&typeMap != nullptr) {
```

Comment:
I think this can be removed now, right?

Reply:
You mean the comment and not the passes right?

Reply:
Yeah